sklearn/compose/_column_transformer.py

```diff
@@ -686,8 +687,22 @@ def _get_transformer_list(estimators):
     Construct (name, trans, column) tuples from list
 
     """
-    transformers = [trans[1] for trans in estimators]
-    columns = [trans[0] for trans in estimators]
+    _estimators = []
+    for tup in estimators:
```

Comment:
right now a warning is given for every step, right? I would maybe just go through, check if the deprecated behavior was used anywhere and then warn once and possibly change the order / define a variable to index into ``trans``.
right now we're supporting a mix of orders within the same call, which is a bit weird to me.

Reply:
I guess if we don't want to accept a mixture, checking the first tuple would be enough.

Reply:
Not in the case I commented below, right?